static/profile/js/profile.js

```diff
@@ -0,0 +1,486 @@
+    $('#pe_timeformat').text(serverSettings.settings.timeFormat+'h');
+    $('#pe_title').text(serverSettings.settings.customTitle);
+
+    var lastvalidfrom = new Date(mongoprofiles[1] && mongoprofiles[1].startDate ? mongoprofiles[1].startDate : null);
+    
+    //timepicker
+    $('#pe_date').on('change', dateChanged);
+    $('#pe_time').on('change', dateChanged);
+
+  
+    // Set values from profile to html
+    fillTimeRanges();
+    // hide unused style of ratios
+    switchStyle();
+    // show proper submit button
+    dateChanged();
+    
+    // date of last record
+    if (lastvalidfrom) {
+      $('#pe_lastrecvalidfrom').html('Last record date: '+lastvalidfrom.toLocaleString()+' <i>(Date must be newer to create new record or the same to update current record)</i>');
+    } else  {
+      $('#pe_lastrecvalidfrom').html('');
+    }
+    console.log('Done initeditor()');
+  }
+  
+  // Handling valid from date change
+  function dateChanged(event) {
+    var newdate = new Date(Nightscout.client.utils.mergeInputTime($('#pe_time').val(), $('#pe_date').val()));
+    if (mongoprofiles.length<2 || !mongoprofiles[1].startDate || mongoprofiles.length>=2 && new Date(mongoprofiles[1].startDate).getTime() === newdate.getTime()) {
+      $('#pe_submit').text('Update record').css('display','');
+      $('#pe_time').css({'background-color':'white'});
+      $('#pe_date').css({'background-color':'white'});
+      $('#pe_submit').css({'background-color':'buttonface'});
+    } else if (mongoprofiles.length<2 || new Date(mongoprofiles[1].startDate).getTime() < newdate.getTime()) {
+      $('#pe_submit').text('Create new record').css('display','');
+      $('#pe_time').css({'background-color':'green'});
+      $('#pe_date').css({'background-color':'green'});
+      $('#pe_submit').css({'background-color':'green'});
+    } else {
+      $('#pe_submit').css('display','none');
+      $('#pe_time').css({'background-color':'red'});
+      $('#pe_date').css({'background-color':'red'});
+      $('#pe_submit').css({'background-color':'red'});
+    }
+    if (event) {
+      event.preventDefault();
+    }
+  }
+  
+  // Handling html events and setting/getting values
+  function switchStyle(event) {
+    if (!$('#pe_perGIvalues').is(':checked')) {
+      $('#pe_simple').show('slow');
+      $('#pe_advanced').hide('slow');
+    } else {
+      $('#pe_simple').hide('slow');
+      $('#pe_advanced').show('slow');
+    }
+    if (event) {
+      event.preventDefault();
+    }
+  }
+  
+  function fillTimeRanges(event) {
+    var mmoltime = [
+      '0:00','0:30','1:00','1:30','2:00','2:30','3:00','3:30','4:00','4:30','5:00','5:30',
+      '6:00','6:30','7:00','7:30','8:00','8:30','9:00','9:30','10:00','10:30','11:00','11:30',
+      '12:00','12:30','13:00','13:30','14:00','14:30','15:00','15:30','16:00','16:30','17:00','17:30',
+      '18:00','18:30','19:00','19:30','20:00','20:30','21:00','21:30','22:00','22:30','23:00','23:30'
+    ];
+    var mgtime = [
```

Comment:
mmoltime and mgtime are pretty misleading as variable names, as the units being used doesn't have anything to do with whether the user prefers to be on a 24 or 12 hour clock and/or if the units being used is mmol or mgdl. Not a biggie but potentially confusing.